diff --git a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
index a9d8177..55586f2 100644
--- a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
+++ b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaProviderComponentView.mm
@@ -5,7 +5,7 @@
 #import <react/renderer/components/safeareacontext/Props.h>
 #import <react/renderer/components/safeareacontext/RCTComponentViewHelpers.h>
 
-#import "RCTFabricComponentsPlugins.h"
+#import <React/RCTFabricComponentsPlugins.h>
 #import "RNCSafeAreaUtils.h"
 
 using namespace facebook::react;
diff --git a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaViewComponentView.mm b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaViewComponentView.mm
index a2623ad..bfb2e3c 100644
--- a/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaViewComponentView.mm
+++ b/node_modules/react-native-safe-area-context/ios/Fabric/RNCSafeAreaViewComponentView.mm
@@ -6,8 +6,8 @@
 #import <react/renderer/components/safeareacontext/RNCSafeAreaViewComponentDescriptor.h>
 #import <react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.h>
 
-#import "RCTConversions.h"
-#import "RCTFabricComponentsPlugins.h"
+#import <React/RCTConversions.h>
+#import <React/RCTFabricComponentsPlugins.h>
 #import "RNCSafeAreaProviderComponentView.h"
 #import "RNCSafeAreaUtils.h"
 
diff --git a/node_modules/react-native-safe-area-context/ios/RNCSafeAreaShadowView.m b/node_modules/react-native-safe-area-context/ios/RNCSafeAreaShadowView.m
index 1597bc8..109fec1 100644
--- a/node_modules/react-native-safe-area-context/ios/RNCSafeAreaShadowView.m
+++ b/node_modules/react-native-safe-area-context/ios/RNCSafeAreaShadowView.m
@@ -1,5 +1,5 @@
+#import <Yoga/Yoga-umbrella.h>
 #import "RNCSafeAreaShadowView.h"
-
 #import <React/RCTAssert.h>
 #include <math.h>
 
