diff --git a/node_modules/react-native/Libraries/Text/Text/RCTTextShadowView.m b/node_modules/react-native/Libraries/Text/Text/RCTTextShadowView.m
index 5e39196..ee3c4af 100644
--- a/node_modules/react-native/Libraries/Text/Text/RCTTextShadowView.m
+++ b/node_modules/react-native/Libraries/Text/Text/RCTTextShadowView.m
@@ -10,7 +10,7 @@
 #import <React/RCTBridge.h>
 #import <React/RCTShadowView+Layout.h>
 #import <React/RCTUIManager.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "NSTextStorage+FontScaling.h"
 #import <React/RCTTextView.h>
diff --git a/node_modules/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m b/node_modules/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m
index 19d202c..4e6a38b 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m
+++ b/node_modules/react-native/Libraries/Text/TextInput/RCTBaseTextInputShadowView.m
@@ -10,7 +10,7 @@
 #import <React/RCTBridge.h>
 #import <React/RCTShadowView+Layout.h>
 #import <React/RCTUIManager.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "NSTextStorage+FontScaling.h"
 #import <React/RCTBaseTextInputView.h>
diff --git a/node_modules/react-native/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m b/node_modules/react-native/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m
index 8322bfc..c755ac5 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m
+++ b/node_modules/react-native/Libraries/Text/TextInput/RCTInputAccessoryShadowView.m
@@ -14,6 +14,7 @@
 - (void)insertReactSubview:(RCTShadowView *)subview atIndex:(NSInteger)atIndex
 {
   [super insertReactSubview:subview atIndex:atIndex];
+
   subview.width = (YGValue) { RCTScreenSize().width, YGUnitPoint };
 }
 
diff --git a/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.m b/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.m
index e83bd8f..52a990d 100644
--- a/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.m
+++ b/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.m
@@ -8,7 +8,7 @@
 #import <React/RCTVirtualTextShadowView.h>
 
 #import <React/RCTShadowView+Layout.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import <React/RCTRawTextShadowView.h>
 
diff --git a/node_modules/react-native/React/Base/RCTConvert.h b/node_modules/react-native/React/Base/RCTConvert.h
index c4913f6..b08d61f 100644
--- a/node_modules/react-native/React/Base/RCTConvert.h
+++ b/node_modules/react-native/React/Base/RCTConvert.h
@@ -14,7 +14,7 @@
 #import <React/RCTLog.h>
 #import <React/RCTPointerEvents.h>
 #import <React/RCTTextDecorationLineType.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 #if TARGET_OS_IPHONE
 #import <WebKit/WebKit.h>
 #endif
diff --git a/node_modules/react-native/React/Views/RCTLayout.h b/node_modules/react-native/React/Views/RCTLayout.h
index 55340ad..b42c31e 100644
--- a/node_modules/react-native/React/Views/RCTLayout.h
+++ b/node_modules/react-native/React/Views/RCTLayout.h
@@ -8,7 +8,7 @@
 #import <UIKit/UIKit.h>
 
 #import <React/RCTDefines.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
diff --git a/node_modules/react-native/React/Views/RCTLayout.m b/node_modules/react-native/React/Views/RCTLayout.m
index e11d7e4..38b8572 100644
--- a/node_modules/react-native/React/Views/RCTLayout.m
+++ b/node_modules/react-native/React/Views/RCTLayout.m
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "RCTAssert.h"
 #import "RCTShadowView+Layout.h"
diff --git a/node_modules/react-native/React/Views/RCTShadowView+Layout.m b/node_modules/react-native/React/Views/RCTShadowView+Layout.m
index 6595f4e..1be893f 100644
--- a/node_modules/react-native/React/Views/RCTShadowView+Layout.m
+++ b/node_modules/react-native/React/Views/RCTShadowView+Layout.m
@@ -7,7 +7,7 @@
 
 #import "RCTShadowView+Layout.h"
 
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "RCTAssert.h"
 
diff --git a/node_modules/react-native/React/Views/RCTShadowView.h b/node_modules/react-native/React/Views/RCTShadowView.h
index 428d2eb..5510dfe 100644
--- a/node_modules/react-native/React/Views/RCTShadowView.h
+++ b/node_modules/react-native/React/Views/RCTShadowView.h
@@ -10,7 +10,7 @@
 #import <React/RCTComponent.h>
 #import <React/RCTLayout.h>
 #import <React/RCTRootView.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 @class RCTRootShadowView;
 @class RCTSparseArray;
diff --git a/node_modules/react-native/React/Views/SafeAreaView/RCTSafeAreaShadowView.m b/node_modules/react-native/React/Views/SafeAreaView/RCTSafeAreaShadowView.m
index a9a7bee..89e72d9 100644
--- a/node_modules/react-native/React/Views/SafeAreaView/RCTSafeAreaShadowView.m
+++ b/node_modules/react-native/React/Views/SafeAreaView/RCTSafeAreaShadowView.m
@@ -8,7 +8,7 @@
 #import "RCTSafeAreaShadowView.h"
 
 #import <React/RCTAssert.h>
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "RCTSafeAreaViewLocalData.h"
 
diff --git a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m
index c892556..913f89e 100644
--- a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m
+++ b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m
@@ -7,7 +7,7 @@
 
 #import "RCTScrollContentShadowView.h"
 
-#import <yoga/Yoga.h>
+#import <Yoga/Yoga-umbrella.h>
 
 #import "RCTUtils.h"
 
diff --git a/node_modules/react-native/ReactCommon/yoga/Yoga-umbrella.h b/node_modules/react-native/ReactCommon/yoga/Yoga-umbrella.h
new file mode 100644
index 0000000..2e62c47
--- /dev/null
+++ b/node_modules/react-native/ReactCommon/yoga/Yoga-umbrella.h
@@ -0,0 +1,21 @@
+#ifdef __OBJC__
+#import <UIKit/UIKit.h>
+#else
+#ifndef FOUNDATION_EXPORT
+#if defined(__cplusplus)
+#define FOUNDATION_EXPORT extern "C"
+#else
+#define FOUNDATION_EXPORT extern
+#endif
+#endif
+#endif
+
+#import "yoga/YGEnums.h"
+#import "yoga/YGMacros.h"
+#import "yoga/YGNode.h"
+#import "yoga/YGStyle.h"
+#import "yoga/YGValue.h"
+#import "yoga/Yoga.h"
+
+FOUNDATION_EXPORT double yogaVersionNumber;
+FOUNDATION_EXPORT const unsigned char yogaVersionString[];
diff --git a/node_modules/react-native/ReactCommon/yoga/Yoga.modulemap b/node_modules/react-native/ReactCommon/yoga/Yoga.modulemap
new file mode 100644
index 0000000..54ea792
--- /dev/null
+++ b/node_modules/react-native/ReactCommon/yoga/Yoga.modulemap
@@ -0,0 +1,6 @@
+framework module yoga {
+  umbrella header "Yoga-umbrella.h"
+
+  export *
+  module * { export * }
+}
diff --git a/node_modules/react-native/ReactCommon/yoga/Yoga.podspec b/node_modules/react-native/ReactCommon/yoga/Yoga.podspec
index bdaefc3..8552570 100644
--- a/node_modules/react-native/ReactCommon/yoga/Yoga.podspec
+++ b/node_modules/react-native/ReactCommon/yoga/Yoga.podspec
@@ -47,11 +47,25 @@ Pod::Spec.new do |spec|
 
   # Set this environment variable when *not* using the `:path` option to install the pod.
   # E.g. when publishing this spec to a spec repo.
-  source_files = 'yoga/**/*.{cpp,h}'
-  source_files = File.join('ReactCommon/yoga', source_files) if ENV['INSTALL_YOGA_WITHOUT_PATH_OPTION']
+  source_files = 'yoga/**/*.{cpp,h}', 'Yoga-umbrella.h'
+  source_files = source_files.map { |file| File.join('ReactCommon/yoga', file) } if ENV['INSTALL_YOGA_WITHOUT_PATH_OPTION']
   spec.source_files = source_files
 
-  header_files = 'yoga/{Yoga,YGEnums,YGMacros,YGNode,YGStyle,YGValue}.h'
-  header_files = File.join('ReactCommon/yoga', header_files) if ENV['INSTALL_YOGA_WITHOUT_PATH_OPTION']
-  spec.public_header_files = header_files
+  spec.module_map = 'Yoga.modulemap'
+
+  # CocoaPods custom `module_map` + `header_dir` doesn't put the umbrella header in right place.
+  # Ideally, modulemap should be placed with the umbrella header, that would work for both use_frameworks! mode and non use_frameworks! mode.
+  # This script copy the umbrella header back to right place.
+  spec.script_phase = {
+    :name => 'Copy umbrella header',
+    :input_files => ["$PODS_ROOT/Headers/Public/Yoga/yoga/Yoga-umbrella.h"],
+    :output_files => ["$PODS_ROOT/Headers/Private/Yoga/Yoga-umbrella.h"],
+    :execution_position => :before_compile,
+    :shell_path => '/bin/bash',
+
+    # In use_frameworks! mode, the umbrella header will by copied to right place.
+    # This copy command will fail because "$PODS_ROOT/Headers/Public/Yoga/yoga/Yoga-umbrella.h" doesn't exist.
+    # The command is just a no-op in use_frameworks! mode.
+    :script => 'cp -f "$PODS_ROOT/Headers/Public/Yoga/yoga/Yoga-umbrella.h" "$PODS_ROOT/Headers/Private/Yoga/Yoga-umbrella.h" || true',
+  }
 end
diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
new file mode 100644
index 0000000..361f5fb
--- /dev/null
+++ b/node_modules/react-native/scripts/.packager.env
@@ -0,0 +1 @@
+export RCT_METRO_PORT=8081
diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index f2ceeda..834514d 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -267,12 +267,12 @@ def modify_flags_for_new_architecture(installer, cpp_flags)
           config_file.save_as(xcconfig_path)
       end
   end
-  # Add RCT_NEW_ARCH_ENABLED to Pods project xcconfig
-  installer.pods_project.targets.each do |target|
-    # if target.name == 'React-Core'
-    if target.name == 'React-Core'
-      puts "#{target.name}"
-      target.build_configurations.each do |config|
+
+  # Add RCT_NEW_ARCH_ENABLED to generated pod target projects
+  installer.target_installation_results.pod_target_installation_results
+    .each do |pod_name, target_installation_result|
+    if pod_name == 'React-Core'
+      target_installation_result.native_target.build_configurations.each do |config|
         config.build_settings['OTHER_CPLUSPLUSFLAGS'] = cpp_flags
       end
     end
